### PR TITLE
Adding GH action workflows for building flannel and flannel kube-proxy images

### PR DESCRIPTION
**Pull request by marosset**
**Reason for PR**:

- Adds a GH action for building new versions of the flannel image
- Updates the GH action used to build kube-proxy images to also build them for flannel CNI (instead of just calico)

Adding GH 
**Issue Fixed**:

Partially fixies #317 

**Requirements**

- [ ] Sqaush commits 
- [ ] Documentation
- [ ] Tests

**Notes**:

/assign @Mik4sa

### CODE REVIEW

File: .github/workflows/build-flannel-images.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow.json
+name: Build Flannel images
+
+on:
+  workflow_dispatch:
```
Comment (Mik4sa) [inline .github/workflows/build-flannel-images.yml:5]: Should we add any other triggers?
What about on push to the default branch?

Am I right that workflow_dispatch means that it must be triggered manually?
If so, when will we trigger a manually build?

File: .github/workflows/build-flannel-images.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build and push images
+      run: |
+        echo "${{ secrets.DOCKER_SECRET }}" | docker login -u ${{ secrets.DOCKER_USERNAME }} --password-stdin
+        cd ./hostprocess/calico && ./build.sh -f ${{ github.event.inputs.flannel_version }}
```
Comment (Mik4sa) [inline .github/workflows/build-flannel-images.yml:19]: Shouldn't the directory be "flannel" instead of "calico"?

Comment (marosset) [reply .github/workflows/build-flannel-images.yml:19]: yes -_-

File: .github/workflows/build-flannel-images.yml (hunk 1 above)
Comment (marosset) [reply .github/workflows/build-flannel-images.yml:5]: Yes `workflow_dispatch` means triggered manually.

If we push on default branch we'd need another versioning strategy.
Right now the flannel version is used for the container image tag / version too and I think should avoid changing the contents of images we already pushed. If we push every time there are updates the default branch (even if we limit the job to changes under hostprocess/flannel) we would be updating the flannel images in between flannel releases.

Let's discuss triggers in a separate issue?

Comment (Mik4sa) [reply .github/workflows/build-flannel-images.yml:5]: Let's do so :)

**Comment by Mik4sa** [comment]
@marosset Otherwise the changes looks good and we could merge IMO. Looks like I can't do much here sadly 😅

**Comment by jsturtevant** [comment]
@Mik4sa thanks for the review!
/lgtm

**Pull request merged**